e2e-tests/gef/cypress/e2e/facility/bank-review-date.spec.js

```diff
@@ -0,0 +1,250 @@
+      bankReviewDate.bankReviewDateMonth().clear().type(todayMonth);
+      bankReviewDate.bankReviewDateYear().clear().type(todayYear);
+
+      bankReviewDate.continueButton().click();
+      bankReviewDate.errorSummary().should('not.exist');
+    });
+
+    it('validates bank review date is less than 6 years in the future', () => {
+      cy.visit(relative(`/gef/application-details/${application.id}/facilities/${facilityId}/bank-review-date`));
+
+      bankReviewDate
+        .bankReviewDateDay()
+        .clear()
+        .type(today.getDate() + 1);
+      bankReviewDate
+        .bankReviewDateMonth()
+        .clear()
+        .type(today.getMonth() + 1);
+      bankReviewDate
+        .bankReviewDateYear()
+        .clear()
+        .type(today.getFullYear() + 7);
+
+      bankReviewDate.continueButton().click();
+      bankReviewDate.errorSummary();
+      bankReviewDate.bankReviewDateError();
+    });
+
+    it('redirects the user to `provided facility` page when form has been successfully filled in', () => {
+      cy.visit(relative(`/gef/application-details/${application.id}/facilities/${facilityId}/bank-review-date`));
+
+      bankReviewDate.bankReviewDateDay().clear().type(todayDay);
+      bankReviewDate.bankReviewDateMonth().clear().type(todayMonth);
+      bankReviewDate.bankReviewDateYear().clear().type(nextYear);
+
+      bankReviewDate.continueButton().click();
+
+      cy.url().should('eq', relative(`/gef/application-details/${application.id}/facilities/${facilityId}/provided-facility`));
+    });
+
+    it('stores the inputted values', () => {
+      cy.visit(relative(`/gef/application-details/${application.id}/facilities/${facilityId}/bank-review-date`));
+
+      bankReviewDate.bankReviewDateDay().clear().type(todayDay);
+      bankReviewDate.bankReviewDateMonth().clear().type(todayMonth);
+      bankReviewDate.bankReviewDateYear().clear().type(nextYear);
+
+      bankReviewDate.continueButton().click();
+
+      cy.visit(relative(`/gef/application-details/${application.id}/facilities/${facilityId}/bank-review-date`));
+      bankReviewDate.bankReviewDateDay().should('have.value', today.getDate());
+      bankReviewDate.bankReviewDateMonth().should('have.value', today.getMonth() + 1);
+      bankReviewDate.bankReviewDateYear().should('have.value', today.getFullYear() + 1);
+    });
+
+    it('redirects to the Application Details page when using facility end date ', () => {
+      cy.visit(relative(`/gef/application-details/${application.id}/facilities/${facilityId}/about-facility`));
+      aboutFacility.facilityName().clear().type('Name');
+      aboutFacility.shouldCoverStartOnSubmissionYes().click();
+      aboutFacility.coverEndDateDay().clear().type(todayDay);
+      aboutFacility.coverEndDateMonth().clear().type(todayMonth);
+      aboutFacility.coverEndDateYear().clear().type(nextYear);
+      aboutFacility.isUsingFacilityEndDateYes().click();
+      aboutFacility.continueButton().click();
```

Comment:
Not a blocker or familiar with the latest patterns/approaches, but some cpyress commands could be nice here, relating to previous comment; It could accept some params with default values, then the test itself is more DRY and less lines e.g

```js
const completeAndSubmitFacilityAboutForm = ({
  name = 'Facility name',
  day = todayDay,
  month = todayMonth,
}) => {
  aboutFacility.coverEndDateDay().clear().type(todayDay);
  aboutFacility.coverEndDateMonth().clear().type(todayMonth);
  // etc
  aboutFacility.continueButton().click();
};
```